crates/recursor/src/recursor_pool.rs

```diff
@@ -48,8 +48,11 @@ pub(crate) struct RecursorPool<P: RuntimeProvider + Send + 'static> {
     active_requests: Arc<Mutex<HashMap<Query, SharedLookup>>>,
 }
 
-impl RecursorPool<TokioRuntimeProvider> {
-    pub(crate) fn from(zone: Name, ns: GenericNameServerPool<TokioRuntimeProvider>) -> Self {
+impl<P> RecursorPool<P>
+where
+    P: RuntimeProvider + Send + 'static,
```

Comment:
I was able to drop the `Send` and `'static` bounds here and in the struct definition as well.